Orange/projection/pca.py

```diff
@@ -45,10 +46,21 @@ def randomized_pca(A, n_components, n_oversamples=10, n_iter="auto",
 
     n_samples, n_features = A.shape
 
-    c = np.atleast_2d(A.mean(axis=0))
+    # Use `mean_variance_axis` instead of `A.sum` as it is much more memory
+    # efficient than scipy's implementation. See GH
+    # https://github.com/scikit-learn/scikit-learn/pull/12841#issuecomment-460238233
+    if sp.issparse(A):
+        means, _ = mean_variance_axis(A, axis=0)
+    else:
+        means = np.mean(A, axis=0)
```

Comment:
We already have `Orange.statistics.util.mean` which is supposed to handle dense and sparse matrices, but it does not have the axis parameter (unlike most other functions in that module). Maybe you could improve that function instead and use it here?

Reply:
That's a much better idea. Unfortunately, `mean_variance_axis` just ignores NaNs and provides no feedback if the data had any NaNs. This means that `mean` wouldn't properly handle NaNs and we'd have no good way of knowing where they occurred.

However, changing `nanmean` to use this is completely fine since this is exactly the behaviour we want there. So I did that.